File: App/PeopleMgt/Models/PageMetadata.cs
namespace PeopleMgt.Models
{
    /// <summary>
    /// Keeps state of paging request between client and server.
    /// </summary>
    public class PageMetadata
    {
        public int Current { get; set; }
        public bool Previous { get; set; }
        public bool Next { get; set; }
        public int Size { get; set; }
        public int PageCount { get; set; }
        public int RecordCount { get; set; }
    }
}

File: App/PeopleMgt/ClientApp/src/app/model/PageMetadata.ts
export class PageMetadata {
  public Current: string;
  public Previous: boolean;
  public Next: boolean;
  public Size: string;
  public PageCount: number;
  public RecordCount: number;
  
  public constructor(init?: Partial<PageMetadata>) {
    (<any>Object).assign(this, init);    
  }
}
